zerver/lib/narrow_helpers.py

```diff
@@ -21,14 +21,15 @@
 import os
 from collections.abc import Collection, Sequence
 from dataclasses import dataclass, field
+from typing import Any
 
 from django.conf import settings
 
 
 @dataclass
 class NarrowTerm:
     operator: str
-    operand: str
+    operand: Any
```

Comment:
I don't think this is a good change; just type is as `str | List[int]`.

Reply:
This suffices to avoid that:

``` diff
diff --git a/zerver/lib/narrow_helpers.py b/zerver/lib/narrow_helpers.py
index 1dc31a6914..c135205056 100644
--- a/zerver/lib/narrow_helpers.py
+++ b/zerver/lib/narrow_helpers.py
@@ -28,13 +28,14 @@ from django.conf import settings
 @dataclass
 class NarrowTerm:
     operator: str
-    operand: str
+    operand: str | int | list[int]
     negated: bool
 
 
 @dataclass
 class NeverNegatedNarrowTerm(NarrowTerm):
     negated: bool = field(default=False, init=False)
+    operand: str
 
 
 def narrow_dataclasses_from_tuples(
```